src/Diagnostics/Diagnostics.jl

```diff
@@ -0,0 +1,19 @@
+module Diagnostics
+
+export
+    run_diagnostic,
+    HorizontalAverage, Timeseries, FieldMaximum, CFL, AdvectiveCFL, DiffusiveCFL, NaNChecker
+
+using Oceananigans,
+      Oceananigans.Operators
+
+using Oceananigans: AbstractDiagnostic
```

Comment:
Should `AbstractDiagnostic` be introduced within this submodule, rather than in `Oceananigans`?

Reply:
Hmmm, I do it in this case because it's used by a shared utility defined in `utils.jl`. Might make sense to move the shared util into the `Diagnostics` or `OutputWriters` module, whichever is defined first so it can be imported by the other.